inc/settings.php

```diff
@@ -49,19 +49,19 @@ function update_privacy_policy_page() {
 	if (
 		// Validate the nonce.
 		! isset( $_POST['_altis_privacy_policy_page_nonce'] ) ||
-		! wp_verify_nonce( sanitize_text_field( $_POST['_altis_privacy_policy_page_nonce'] ), 'altis.privacy_policy_page' ) ||
+		! wp_verify_nonce( sanitize_text_field( wp_unslash( $_POST['_altis_privacy_policy_page_nonce'] ) ), 'altis.privacy_policy_page' ) ||
 		// Bail early if we're not on the consent page.
 		! isset( $_POST['option_page'] ) ||
-		'cookie_consent_options' !== esc_attr( $_POST['option_page'] )
+		'cookie_consent_options' !== sanitize_text_field( wp_unslash( $_POST['option_page'] ) )
 	) {
 		return;
 	}
 
 	$privacy_option         = 'wp_page_for_privacy_policy';
 	$privacy_policy_page_id = (int) get_option( $privacy_option );
-	$updated_id             = esc_attr( $_POST[ $privacy_option ] );
+	$updated_id             = ! empty( $_POST[ $privacy_option ] ) ?? sanitize_text_field( wp_unslash( $_POST[ $privacy_option ] ) );
```

Comment:
Think I missed this before, the null coalesce isn't going to do anything here because `! empty()` will always return true or false, never `null`. You can remove everything after the `! empty()` check.

Reply:
👍
I think that changes the intent I was going for. I had originally wrote this as 

```php
$updated_id = ! empty( $_POST[ $privacy_option ] ) ? sanitize_text_field( wp_unslash( $_POST[ $privacy_option ] ) ) : '';
```

...but I thought maybe the null coalesce might work here since I was assigning the variable to an empty string if the $_POST variable was empty. I'm still figuring out where null coalesce works and where it doesn't.

Reply:
You're probably looking for the shorthand ternary in that case eg. `?:`. If the expression before it is falsy you'll get false, if not you'll get the value after it.

Reply:
Yep. That's the one. In my head I'm still fuzzy on the differences, aside from that I've used both at different times. 👍

Reply:
Null coalesce is way of assigning a fallback value if the expression before `??` resolves as `null` - `false` is not `null`, it's a boolean value. You'd expect `null` if a variable was undefined for example, this makes it useful for checking multiple levels deep in an array or object in one go, or checking a nullable function argument and providing a fallback.